### PR TITLE
GNOME Adwaita dark window decorations

### PR DESCRIPTION
**Pull request by wigust**
GNOME Adwaita dark window decorations

emacs-dark
emacs-light

### CODE REVIEW

File: layers/+themes/dark-window/README.org [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ - [[#install][Install]]
+ - [[#screenshots][Screenshots]]
+
+* Description
+Gnome 3's default theme Adwaita has support for both light and dark window
+decorations. Many media apps like Totem use the dark theme. It puts the focus
+on the content and it's easier on the eyes.
+
+Emacs with a dark theme and the large white-ish window decoration isn't really a
+fit on my desktop, but that can be easily fixed using a bit of elisp with xprop!
```
Comment (bmag) [inline layers/+themes/dark-window/README.org]: This is a direct quote from the article, so it should be formatted in a quote block. Don't plagiarize.

File: layers/+themes/dark-window/README.org [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+See also:
```
Comment (bmag) [inline layers/+themes/dark-window/README.org]: This is not a "see also", it's the source of a quote.

File: layers/+themes/dark-window/packages.el [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+;;
+;;; License: GPLv3
+
+(defconst dark-window-packages
+  '(frame-cmds))
+
+(defun dark-window/init-frame-cmds ()
+  (use-package frame-cmds
+    :config
+    (progn
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: This whole `progn` is a direct copy of http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html, so you should a comment attributing credit to the original author of the code.

File: layers/+themes/dark-window/packages.el [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      (defun set-selected-frame-dark ()
+        (interactive)
+        (let ((frame-name (get-frame-name (selected-frame))))
+          (call-process-shell-command
+           (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"dark\" -name \"" frame-name "\""))))
+
+      (if (window-system)
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: Use `when` when you don't have an `else` clause.

**Comment by wigust** [comment]
@bmag thanks. I added author and a quotation block. Also added a reason of creation in the description. It's cannot be fixed easily in GNOME like in other GNOME apps.

**Comment by bmag** [comment]
I clicked the approve button because you fixed most of the issues, but you still need to add a comment before that `progn`.

**Comment by wigust** [comment]
@bmag OK, added.

File: layers/+themes/dark-window/packages.el [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  '(frame-cmds))
+
+(defun dark-window/init-frame-cmds ()
+  (use-package frame-cmds
+    :config
+    ;; Based on Nicolas Petton code http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html
+    (progn
+      (defun set-selected-frame-color ()
+        (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-dark))
+          (interactive)
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: `(interactive)` must be the first sexp in the function's body (after the docstring).

File: layers/+themes/dark-window/packages.el [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+(defconst dark-window-packages
+  '(frame-cmds))
+
+(defun dark-window/init-frame-cmds ()
+  (use-package frame-cmds
+    :config
+    ;; Based on Nicolas Petton code http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html
+    (progn
+      (defun set-selected-frame-color ()
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: Please a documentation string for the function.

Comment (bmag) [inline layers/+themes/dark-window/packages.el]: Need to find a better name. The function doesn't set an arbitrary color, it specifically sets the decoration to dark. In fact, we'd probably want an argument to decide if the decoration should set to dark or light.

File: layers/+themes/dark-window/packages.el [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+        (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-dark))
+          (interactive)
+          (let ((frame-name (get-frame-name (selected-frame))))
+            (call-process-shell-command
+             (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"dark\" -name \"" frame-name "\"")))))
+
+      (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-light))
+        (interactive)
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: `(interactive)` doesn't do anything here and should be removed.

File: layers/+themes/dark-window/packages.el [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    ;; Based on Nicolas Petton code http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html
+    (progn
+      (defun set-selected-frame-color ()
+        (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-dark))
+          (interactive)
+          (let ((frame-name (get-frame-name (selected-frame))))
+            (call-process-shell-command
+             (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"dark\" -name \"" frame-name "\"")))))
+
+      (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-light))
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: Why only if the theme is `spacemacs-light`? There might be a programmatic way to find out if the current theme has a dark or a light background.

File: layers/+themes/dark-window/packages.el [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+        (interactive)
+        (let ((frame-name (get-frame-name (selected-frame))))
+          (call-process-shell-command
+           (concat "xprop -f _GTK_THEME_VARIANT 8u -set _GTK_THEME_VARIANT \"light\" -name \"" frame-name "\"")))))))
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: Why not call `set-selected-frame-color` with an argument that will make the decoration light?

File: layers/+themes/dark-window/packages.el [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+(defconst dark-window-packages
+  '(frame-cmds))
+
+(defun dark-window/init-frame-cmds ()
+  (use-package frame-cmds
+    :config
+    ;; Based on Nicolas Petton code http://nicolas.petton.fr/blog/emacs-dark-window-decoration.html
+    (progn
+      (defun set-selected-frame-color ()
+        (when (and (eq window-system 'x) (eq spacemacs--cur-theme 'spacemacs-dark))
```
Comment (bmag) [inline layers/+themes/dark-window/packages.el]: The theme shouldn't matter here.

**Comment by wigust** [comment]
Cannot find any variable or function which will tell me if theme or background is dark/light
For some reason color of borders doesn't change for **first** make-frame.

**Pull request closed without merging**